src/Microsoft.OData.Edm/Schema/CoreModel/EdmCoreModelPathType.cs

```diff
@@ -0,0 +1,55 @@
+//---------------------------------------------------------------------
+// <copyright file="EdmCoreModelPathType.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Edm
+{
+    /// <summary>
+    /// The built-in Edm.AnnotationPath, Edm.PropertyPath, Edm.NavigationPropertyPath abstrat type in the core model.
+    /// </summary>
+    internal sealed class EdmCoreModelPathType : EdmType, IEdmPathType, IEdmCoreModelElement
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EdmCoreModelPathType"/> class.
+        /// </summary>
+        /// <param name="name">The name.</param>
+        /// <param name="pathKind">The path kind.</param>
+        public EdmCoreModelPathType(string name, EdmPathTypeKind pathKind)

```

Comment:
>EdmCoreModelPathType [](start = 15, length = 20)

What is the advantage of having a single EdmCoreModelPathType with a name, versus having specific derived types for AnnotationPath, PropertyPath, and NavigationPropertyPath?

Reply:
I reuse the primitive type design pattern.  We don't have Int32, Double, ... sub types derived from EdmCoreModelPrimitiveType.

Otherwise, I provide the methods in the EdmCoreModel.cs likes:

```C#
GetAnnotationPath(bool isNullable)
GetPropertyPath(bool isNullable)
GetNavigationPath(bool isNullable)
```